### PR TITLE
[Sat Extension] Additional acquisition parameters

### PR DESCRIPTION
**Pull request by emmanuelmathot**
**Proposed Changes:**

1. [Sat extension] Additional acquisition parameters

**PR Checklist:**

- [x] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [x] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [ ] This PR affects the [STAC API spec](https://github.com/radiantearth/stac-api-spec), and I have opened issue/PR #XXX to track the change.

### CODE REVIEW

File: extensions/sat/README.md
Comment (m-mohr) [inline extensions/sat/README.md:21]: That seems useful for searching indeed, but why is the name so different from all the other names mentioned in the description?

File: extensions/sat/README.md
Comment (m-mohr) [inline extensions/sat/README.md:23]: We have discussed absolute orbit earlier and didn't find it useful for searching. So if there's not new reasons for inclusion, I'd remove it again.

File: extensions/sat/README.md
Comment (m-mohr) [inline extensions/sat/README.md:21]: That seems indeed useful for search, but why is the name so different from all the names mentioned in the description?

File: extensions/sat/README.md
Comment (m-mohr) [inline extensions/sat/README.md:23]: We've discussed absolute_orbit earlier in issues and PRs and decided against it as it doesn't seem very useful for search. So I'd remove this again unless there are new strong arguments for it.

File: extensions/sat/README.md
Comment (m-mohr) [inline extensions/sat/README.md:26]: It somewhat sounds like these field would be a good fit for the view extension, but not sure. Other than that I'm not exactly sure how useful these fields would be in search. Also, how commonly are these available?

File: extensions/sat/README.md
Comment (m-mohr) [inline extensions/sat/README.md:25]: I'm missing a description what this field is useful for.

File: extensions/sat/json-schema/schema.json
Comment (m-mohr) [inline extensions/sat/json-schema/schema.json:36]: This can be simplified with "minProperties: 1"

**Review by m-mohr — CHANGES_REQUESTED** [review]

File: extensions/sat/README.md
Comment (emmanuelmathot) [reply extensions/sat/README.md:21]: Indeed. I chose an identifier largely used in the industry (cf . No problem using something like `sat:platform_international_designator`. In addition, it shall remove the confusion where serial identifier also designate the  sequence number of the platform among the mission.

File: extensions/sat/README.md
Comment (emmanuelmathot) [reply extensions/sat/README.md:23]: I would have several reasons:

1.  It is a very common parameter in the industry. Often used in the items product identifier thus derivable for search.
2.  In mission planning and tasking, the absolute orbit is often use as a reference. Especially in the non systematic acquisition missions. The resulting product can be tagged with the absolute orbit and thus searchable as such.
3.  In the case of orbital changes during the mission modifying the ground track and thus the repeat cycle and thus the number or relative orbits, the combination of cycle and relative orbit is not sufficient to derive an absolute orbit. This has been the case for ENVISAT mission: To ensure an additional 3 years lifespan, the ENVISAT satellite moved to a new lower orbit on October  22th, 2010. From November 2nd, 2010, for the ENVISAT extension orbit, the ground track changes and consequently the repeat cycle changes: 30 days with 431 orbits per cycle instead of 35 days-501 orbits per cycle.

File: extensions/sat/README.md
Comment (emmanuelmathot) [reply extensions/sat/README.md:25]: This is useful when you want to compute quickly orbital parameters without having to download the product. For instance to compute on the fly a baseline between 2 satellite acquisition This is used to find the best candidate in the archive from a post disaster event (e.g earthquake) scene acquisition for a DInSAR processing.

File: extensions/sat/README.md
Comment (emmanuelmathot) [reply extensions/sat/README.md:26]: Indeed, I have not seen that a similar parameter was present in view extension. Maybe the view should also include the look angle to know if this is along or across track.

**Comment by emmanuelmathot** [comment]
Hello. Can we move on this PR? @m-mohr  Do the reviews need more clarifications?

File: extensions/sat/README.md
Comment (matthewhanson) [reply extensions/sat/README.md:23]: So would users use the absolute orbit for that? It seems like using datetime would be a more natural way to do that. Users would know the dates of the changes and if they corresponded with their dates of interest. Figuring out absolute orbit seems like an unnecessary step.

I'm not opposed to including it, it would be an optional field and I have worked with some datasets that use it explicitly in data organization.

Comment (cholmes) [reply extensions/sat/README.md:23]: I think I'm ok with this in the 'sat extension'. I think we should aim to consolidate fields that are generally useful for satellite data providers, even if they wouldn't be in search. Instead of making providers all come up with their own they should be able to use common ones. We could make some additional extension that is just focused on 'additional satellite properties', but that seems like overkill to me. I feel ok with it here since it's an extension, and an optional field. 

Anything in core should be for search, absolutely, but I think it's ok for extensions to add in fields that aren't strictly for search. Like projection & rendering hints.

File: extensions/sat/README.md
Comment (cholmes) [reply extensions/sat/README.md:25]: @emmanuelmathot - could you just edit the PR to include more of that information? We want the description of 'sat:anx_date_time' to not just say what it is, but a bit of what it is used for. If it's more than a couple lines you can add more information below the table.

File: extensions/sat/README.md
Comment (cholmes) [inline extensions/sat/README.md:38]:
```suggestion
```suggestion
Used to quickly compute orbital parameters without having to download the product. For instance to compute on the fly a baseline between 2 satellite acquisition, to find the best candidate in the archive from a post-disaster event (e.g earthquake) scene acquisition for a DInSAR processing.
```
```

File: extensions/sat/README.md
Comment (cholmes) [reply extensions/sat/README.md:25]: Oh, I see now that you do have it below. Just put a suggestion in for how to edit it.

File: extensions/sat/README.md
Comment (cholmes) [inline extensions/sat/README.md:33]: This could also use a bit more 'why', and also a period at the end.

File: extensions/sat/README.md
Comment (cholmes) [inline extensions/sat/README.md:42]: Is this in the example in the directory? If so then I'd put a link in here too.

**Review by cholmes — CHANGES_REQUESTED** [review]
Overall I think this is looking good. I made a couple comments, but I think the overall thing is just add a bit more 'why' to the descriptions, help people understand what these are used for. But thanks for the contribution, it's great to see these suggestions come from working with the spec and data.

File: extensions/sat/examples/example-sentinel1.json
Comment (m-mohr) [inline extensions/sat/examples/example-sentinel1.json:64]: That field is not part of the SAR extension and should use a different prefix or be removed, same for all assets.

File: extensions/sat/json-schema/schema.json
Comment (m-mohr) [inline extensions/sat/json-schema/schema.json:40]: Minimum doesn't make sense for strings

File: extensions/sat/json-schema/schema.json
Comment (m-mohr) [inline extensions/sat/json-schema/schema.json:59]:
```suggestion
            "sat:anx_datetime": {
```

**Review by m-mohr — CHANGES_REQUESTED** [review]
See comments :-)

File: extensions/sat/README.md
Comment (cholmes) [inline extensions/sat/README.md:43]:
```suggestion
Some documented examples are available for [Sentinel-1](examples/example-sentinel1.json).
```

**Review by cholmes — APPROVED** [review]
Looks good to me, with one minor english tweak. Thanks!

File: extensions/sat/examples/example-sentinel1.json
Comment (m-mohr) [reply extensions/sat/examples/example-sentinel1.json]: I'm unresolving this as there are still sar:swaths in the file.

File: extensions/sat/json-schema/schema.json
Comment (m-mohr) [inline extensions/sat/json-schema/schema.json:56]: Just wondering: Can relative orbit be 0 (or even negative)? Can absolute orbit be 0?

Comment (emmanuelmathot) [reply extensions/sat/json-schema/schema.json:56]: not in my knowledge, it is always starting from 1 in the beginning of the repeat cycle.

File: extensions/sat/examples/example-sentinel1.json
Comment (emmanuelmathot) [reply extensions/sat/examples/example-sentinel1.json]: I don't know how exactly works this review mechanism but there is no more swath in the latest commit

File: extensions/sat/examples/example-sentinel1.json
Comment (emmanuelmathot) [reply extensions/sat/examples/example-sentinel1.json]: done

File: extensions/sat/examples/example-sentinel1.json
Comment (m-mohr) [reply extensions/sat/examples/example-sentinel1.json]: Not sure what happened, but now it's gone for me, too.

**Review by m-mohr — APPROVED** [review]
So it's looks good to be merged in the spec. I'm personally neutral on these changes, but I need to approve after I have requested changes so I'll approve to unblock the merge. Nevertheless, I'd be happy if more data providers could look over the changes and approve them. So I'm not merging myself, although the required number of 2 reviewers have been reached.

File: CHANGELOG.md
Comment (m-mohr) [inline CHANGELOG.md:12]: Better specify explicitly which fields have been added. That helps devs to track down changes.

Comment (emmanuelmathot) [reply CHANGELOG.md]: fields listed in changelog

**Comment by emmanuelmathot** [comment]
> So it's looks good to be merged in the spec. I'm personally neutral on these changes, but I need to approve after I have requested changes so I'll approve to unblock the merge. Nevertheless, I'd be happy if more data providers could look over the changes and approve them. So I'm not merging myself, although the required number of 2 reviewers have been reached.

Ok, so how do we proceed? Is there something we can do?

**Comment by m-mohr** [comment]
@emmanuelmathot Maybe get another "independent" voice that confirms that these changes make sense to them as well and/or they use these fields?

**Comment by cholmes** [comment]
I'm fine to merge this. If it was in core I'd want more independent review, but I'm ok in an extension. I think it's good to provide more optional fields, instead of making everyone come up with their own. If one of the field names ends up totally wrong we can deprecate it and add a new one, without breaking backwards compatability.

**Pull request merged**
